packages/langium/src/documents/document-builder.ts

```diff
@@ -85,14 +85,14 @@ export class DefaultDocumentBuilder implements DocumentBuilder {
     documentChanged(uri: URI): void {
         this.langiumDocuments.invalidateDocument(uri);
         const newDocument = this.langiumDocuments.getOrCreateDocument(uri);
+        this.build(newDocument);
         this.indexManager.update(newDocument);
         this.langiumDocuments.all.filter(doc => this.isAffected(doc, uri)).forEach(
             doc => {
                 this.build(doc);
                 this.indexManager.update(doc);
             }
         );
-        this.build(newDocument);
```

Comment:
This conflicts with the changes in #244. @msujew we should test the domainmodel example thoroughly in your PR.